### ISSUE TITLE
Bug in Estimator tutorial?

### ISSUE THREAD

**Issue by Hvass-Labs** [tensorflow/tensorflow#13565]
This is the tutorial for the Estimator-class:

https://www.tensorflow.org/extend/estimators

You have the following code:

    my_nn = tf.estimator.DNNClassifier(feature_columns=[age, height, weight],
                                       hidden_units=[10, 10, 10],
                                       activation_fn=tf.nn.relu,
                                       dropout=0.2,
                                       n_classes=3,
                                       optimizer="Adam")

and the following:

    input_layer = tf.feature_column.input_layer(
        features=features, feature_columns=[age, height, weight])

If I understand correctly, the feature-columns use the __variables__ `age`, `height` and `weight`. However, these __variables__ are not defined anywhere in the source-code for the tutorial.

The complete source-code is available in `abalone.py` as well:

https://github.com/tensorflow/tensorflow/blob/r1.3/tensorflow/examples/tutorials/estimators/abalone.py

But here you no longer have the `input_layer` that uses the feature-columns. Instead you have the following which pulls out `"x"` from the `features`-dict:

    # Connect the first hidden layer to input layer
    # (features["x"]) with relu activation
    first_hidden_layer = tf.layers.dense(features["x"], 10, activation=tf.nn.relu)

So I'm a bit confused how this is supposed to work?

In general, why don't you make the tutorials as Jupyter Notebooks instead? It would be immensely more helpful than your current tutorial style which is very confusing.

And please remember, that for each hour you spend polishing the code, you will likely save several hours of head-aches for each person trying to understand your code. Multiplied by the many thousands of TensorFlow users, this is a tremendous amount of work-hours that is freed up for the community!

**Comment by yaroslavvb** [comment]
The reason for using files rather than notebooks is probably for testing reasons. IE, [test_tutorials.sh](https://github.com/tensorflow/tensorflow/blob/d040289dcf825fe2ac9579f4b0175a23833a57fa/tensorflow/tools/ci_build/builds/test_tutorials.sh) runs all the files and make sure they don't crash.

**Comment by reedwm** [comment]
@sandersk, can you comment why `input_layer` is not used in the source file?

The issue #11785 suggests having Jupyter Notebook tutorials. Pinging @wolffg on that issue.

**Comment by wolffg** [comment]
We're working on improved estimator docs now.  In the interim, another
source of information is Magnus's blog post:
https://developers.googleblog.com/2017/09/introducing-tensorflow-datasets.html

**Comment by sandersk** [comment]
Looks like the docs may have gotten a bit out of sync with recent code updates to align with the latest version of the Estimator API. As wolffg mentioned, we are in the process of updating the estimator docs now.

**Comment by tensorflowbutler** [comment]
It has been 14 days with no activity and the `awaiting tensorflower` label was assigned. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
It has been 14 days with no activity and the `awaiting tensorflower` label was assigned. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Awaiting TensorFlower: It has been 14 days with no activity and the `awaiting tensorflower` label was assigned. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
A member of the TensorFlow organization has replied after the stat:awaiting tensorflower label was applied.

**Comment by tatatodd** [comment]
Assigning to @wolffg based on the earlier note about working on improved estimator docs.

**Comment by tensorflowbutler** [comment]
Nagging Assignee: It has been 14 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @wolffg: It has been 14 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @wolffg: It has been 14 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @wolffg: It has been 14 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @wolffg: It has been 14 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by wolffg** [comment]
This doc has been replaced by these two files:

https://www.tensorflow.org/versions/master/get_started/custom_estimators
https://github.com/tensorflow/models/blob/master/samples/core/get_started/custom_estimator.py

It is 100% true we should remove the outdated sample still in tensorflow/tensorflow, but outdated sample cleanup is being tracked elsewhere, so I'm closing this.  Thanks for reporting this.
